src/en/hstream/src/eu/kanade/tachiyomi/animeextension/en/hstream/Hstream.kt

```diff
@@ -0,0 +1,440 @@
+        }
+    }
+
+    // Details
+
+    override fun animeDetailsParse(document: Document): SAnime {
+        val anime = SAnime.create()
+        anime.thumbnail_url = baseUrl + document.selectFirst("div.thumbook div.thumb img")!!.attr("src")
+        val title = document.select("div.infox h1").text()
+        anime.title = title
+        anime.genre = document.select("div.info-content div.genxed a")
+            .joinToString(", ") { it.text() }
+        anime.description = document.select("div.info-content div.desc").text().substringAfter("Watch $title on HentaiStream.moe in 720p 1080p and (if available) 2160p (4k).").trim()
+        anime.author = document.select("div.info-content div.spe span:nth-child(2)  a")
+            .joinToString(", ") { it.text() }
+        anime.status = parseStatus(document.select("div.info-content div.spe span:nth-child(1)").text().trim().substringAfter("Status: "))
+        return anime
+    }
+
+    private fun parseStatus(statusString: String): Int {
+        return when (statusString) {
+            "Ongoing" -> SAnime.ONGOING
+            else -> SAnime.COMPLETED
+        }
+    }
+
+    // Latest
+
+    override fun latestUpdatesSelector(): String = "article.bs"
+
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/hentai/search?order=latest&page=$page")
+
+    override fun latestUpdatesFromElement(element: Element): SAnime {
+        val anime = SAnime.create()
+        anime.setUrlWithoutDomain(baseUrl + element.select("a").attr("href"))
+        anime.title = element.select("h2").text()
+        anime.thumbnail_url = baseUrl + element.select("img").attr("src")
+        return anime
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = "ul.pagination li a[rel=next]"
+
+    // Settings
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val videoQualityPref = ListPreference(screen.context).apply {
+            key = PREF_QUALITY_KEY
+            title = PREF_QUALITY_TITLE
+            entries = PREF_QUALITY_ENTRIES
+            entryValues = PREF_QUALITY_ENTRIES
+            setDefaultValue("720p")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(key, entry).commit()
+            }
+        }
+
+        val subLangPref = ListPreference(screen.context).apply {
+            setDefaultValue("English")
+            summary = "%s"
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(key, entry).commit()
+            }
+        }
```

Comment:
what is the point of this?

Reply:
If that site decides to put more subs in future then I can use this function.